resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -99,6 +102,16 @@ private[spark] class ExecutorPodsAllocator(
   @volatile private var deletedExecutorIds = Set.empty[Long]
 
   def start(applicationId: String, schedulerBackend: KubernetesClusterSchedulerBackend): Unit = {
+    // wait until the driver pod is ready to ensure executors can connect to driver svc
```

Comment:
Can we be more specific? The problem is the absence of K8s's headless service resource for this driver pod. For example, since K8s is asynchronously working, the problem can happen even when the driver pod is ready with all sidekicks and the K8s service is not ready to work with this driver pod.

Reply:
Thank you for the review. Makes sense to me. Addressed in the newer commit.